app/controllers/api/framework_assessments_controller.rb

```diff
@@ -38,17 +38,21 @@ def show
   private
 
     def new_assessment_params
-      params.require(:data).permit(NEW_PERMITTED_PER_PARAMS).to_h
+      params.require(:data).permit(NEW_ASSESSMENT_PERMITTED_PARAMS).to_h
     end
 
     def update_assessment_params
-      params.require(:data).permit(UPDATE_PERMITTED_PER_PARAMS)
+      params.require(:data).permit(UPDATE_ASSESSMENT_PERMITTED_PARAMS)
     end
 
     def update_assessment_status
       update_assessment_params.to_h.dig(:attributes, :status)
     end
 
+    def handover_details
```

Comment:
This feels like it should be in the PER controller as it's very specific to that model only. I was thinking that we could pass in the update params to the `confirm!` method instead of just the status as so:
```
assessment.confirm!(update_assessment_details)
```
meaning we just define our custom `update_assessment_params` in the PER controller to permit handover fields. For the confirm method it could look like this: 
```
def confirm!(details)
    return unless details[:status] == ASSESSMENT_CONFIRMED

    state_machine.confirm!
    self.handover_details = details[:handover_details]if details[:handover_details].present? && respond_to?(:handover_details)
   ...
```
or something similar, let me know your thoughts

Reply:
I don't mind that, but the word `details` seems very repetitive. I'll try a variation of this suggestion to move the PER specific stuff out of the generic assessment controller 🤔